### PR TITLE
Rename ONNXRUNTIME_API_STATUSCALL to ONNXRUNTIME_API_CALL since it's …

### PR DESCRIPTION
**Pull request by RyanUnderhill**
…just for calling convention, there's no status return value implied.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/cpu_provider_factory.cc [hunk 1]
```diff
@@ -15,7 +15,7 @@ struct CpuProviderFactory {
   CpuProviderFactory();
 };
 
-ONNXStatus* ONNXRUNTIME_API_STATUSCALL CreateCpu(void* this_, ONNXRuntimeProvider** out) {
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/cpu_provider_factory.cc]: this is why it was called ...STATUSCALL... to differentiate from the regular ONNXRUNTIME_API_CALL invocations.

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 2]
```diff
@@ -79,14 +79,14 @@ typedef void ONNXStatus;
 
 // __VA_ARGS__ on Windows and Linux are different
 #define ONNXRUNTIME_API(RETURN_TYPE, NAME, ...) \
-  ONNX_RUNTIME_EXPORT RETURN_TYPE ONNXRUNTIME_API_STATUSCALL NAME(__VA_ARGS__) NO_EXCEPTION
+  ONNX_RUNTIME_EXPORT RETURN_TYPE ONNXRUNTIME_API_CALL NAME(__VA_ARGS__) NO_EXCEPTION
```
Comment (snnn) [inline include/onnxruntime/core/session/onnxruntime_c_api.h]: Don't change these three lines.

File: onnxruntime/core/providers/cpu/cpu_provider_factory.cc [hunk 3]
```diff
@@ -15,7 +15,7 @@ struct CpuProviderFactory {
...
+ONNXStatus* ONNXRUNTIME_API_CALL CreateCpu(void* this_, ONNXRuntimeProvider** out) {
```
Comment (snnn) [inline onnxruntime/core/providers/cpu/cpu_provider_factory.cc]: For all the function implementations, please change it to 
ONNXRUNTIME_API(...)
or 
ONNXRUNTIME_API_STATUS_IMPL(...)

File: onnxruntime/core/providers/cpu/cpu_provider_factory.cc (hunk 1 above)
Comment (pranavsharma) [reply onnxruntime/core/providers/cpu/cpu_provider_factory.cc]: had offline conversation with Ryan. this is ok.

**Pull request merged**